src/runtime_src/core/pcie/emulation/common_em/config.h

```diff
@@ -89,6 +90,89 @@ namespace xclemulation{
   const int VIVADO_MIN_VERSION = 2000;
   const int VIVADO_MAX_VERSION = 2100;
 
+inline bool file_exists(const std::string& fnm)
+{
+  struct stat statBuf;
+  return stat(fnm.c_str(), &statBuf) == 0;
+}
+
+
+//sparse log utility
+enum class eEmulationType{
+  eSw_Emu,
+  eHw_Emu
+};
+
+struct sParseLog
+{
+  std::ifstream file;
+  std::string mFileName;
+  std::atomic_bool mFileExists;
+  std::vector<std::string> mMatchedStrings;
+  eEmulationType mEmuType;
+
+  sParseLog(const std::string &iDeviceLog, eEmulationType iType, std::vector<std::string> &iMatchedStrings)
+      : mFileName(iDeviceLog), mMatchedStrings(iMatchedStrings), mEmuType(iType)
+  {
+
+    mFileExists.store(false);
+  }
+
+  ~sParseLog() = default;
+
+  void SearchString()
+  {
+    std::string line;
+    while (std::getline(file, line))
+    {
+      for (const auto &StringData : mMatchedStrings)
+      {
+        if (line.find(StringData) != std::string::npos)
+        {
+          if (eEmulationType::eSw_Emu == mEmuType)
+            printMsg();
+          else if (eEmulationType::eHw_Emu == mEmuType)
+          {
+            if (!StringData.compare("Exiting xsim") || !StringData.compare("FATAL_ERROR"))
+              printMsg();
+            else
+              std::cout << line << std::endl;
+          }
+        }
+      }
+    }
+  }
+  void printMsg()
+  {
+    switch (mEmuType)
+    {
+    case eEmulationType::eSw_Emu:
+      std::cout << "Received request to end the application. Press Cntrl+C to exit the application." << std::endl;
+      break;
+    case eEmulationType::eHw_Emu:
+      std::cout << "SIMULATION EXITED" << std::endl;
+      break;
+    }
+  }
+  void parseLog()
+  {
+    if (not mFileExists.load())
+    {
+      if (file_exists(mFileName))
+      {
+        file.open(mFileName);
```

Comment:
What is file?   Again inconsistent naming convention.  Some member data is `mMember` but this one is without `m`.  You are not alone here; I also do this some times only to regret later.

Reply:
The description is added to the function so hopefully it conveys its existence.